lib/iex/lib/iex/server.ex

```diff
@@ -321,10 +344,24 @@ defmodule IEx.Server do
   defp iex_state(opts) do
     prefix = Keyword.get(opts, :prefix, "iex")
     on_eof = Keyword.get(opts, :on_eof, :stop_evaluator)
+    previous_options = Keyword.get(opts, :previous_options, [])
+
+    {previous_options, opts} =
+      cond do
+        prefix == "iex" and Enum.any?(previous_options) ->
```

Comment:
I don't like looking into the prefix to decide this because the prefix can be changed/customized. Is there anything else we can look into options to decide if we should use the previous options or not?

Or maybe we should just evaluate dot-iex again on respawn/continue?

Reply:
Maybe evaluate it again can be a good option. I gonna try this.